### PR TITLE
PG-348 Create an automated daily build process

### PR DESCRIPTION
**Pull request by adivinho**
Screenshot 2022-04-26 at 10 22 22
Screenshot 2022-04-26 at 10 25 39

### CODE REVIEW

File: ppg/pg_stat_monitor-autobuild.groovy [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+
+void installCli(String PLATFORM) {
+    sh """ 
+        set -o xtrace
+        if [ -d aws ]; then
+            rm -rf aws
+        fi
+        if [ ${PLATFORM} = "deb" ]; then
+            sudo apt-get update
+            sudo apt-get -y install wget curl unzip
```
Comment (EvgeniyPatlan) [inline ppg/pg_stat_monitor-autobuild.groovy:15]: I suggest adding gnupg2

File: ppg/pg_stat_monitor-autobuild.groovy [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+            until sudo apt-get -y install gpgv curl clang-11 gnupg; do
+                sleep 30
+                echo "Waiting ..."
+            done
+            sudo wget https://repo.percona.com/apt/percona-release_latest.\$(lsb_release -sc)_all.deb
+            sudo dpkg -i percona-release_latest.\$(lsb_release -sc)_all.deb
+        fi
+
+        sudo percona-release enable ppg-${PG_RELEASE} release
+        pwd -P
```
Comment (EvgeniyPatlan) [inline ppg/pg_stat_monitor-autobuild.groovy:65]: do we need this line?

File: ppg/pg_stat_monitor-autobuild.groovy [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+        export build_dir=\$(pwd -P)
+        set -o xtrace
+        cd \${build_dir}
+        if [ -f ./test/pg-stat-monitor.properties ]; then
+            . ./test/pg-stat-monitor.properties
```
Comment (EvgeniyPatlan) [inline ppg/pg_stat_monitor-autobuild.groovy:70]: better to use source command

File: ppg/pg_stat_monitor-autobuild.groovy [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+                label 'min-bionic-x64'
+            }
+            steps {
+                echo "====> Build docker container on Ubuntu 18.04 PG${PG_RELEASE}"
+                cleanUpWS()
+                installCli("deb")
+                unstash 'properties'
+                popArtifactFolder("rpm/", AWS_STASH_PATH)
+                sh ''' 
+                    ls -la rpm/
```
Comment (EvgeniyPatlan) [inline ppg/pg_stat_monitor-autobuild.groovy:330]: I assume it was for debug

File: ppg/pg_stat_monitor-autobuild.groovy [hunk 5]
```diff
@@ -0,0 +1,382 @@
...
+                    sudo apt-get install -y apt-transport-https ca-certificates curl gnupg-agent software-properties-common
+                    sudo curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+                    sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu bionic stable"
+                    sudo apt-get update
+                    sudo apt-get install -y docker-ce docker-ce-cli containerd.io
```
Comment (EvgeniyPatlan) [inline ppg/pg_stat_monitor-autobuild.groovy:335]: we can use docker.io package from default repo

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]

File: ppg/pg_stat_monitor-autobuild.groovy (hunk 4 above)
Comment (adivinho) [reply ppg/pg_stat_monitor-autobuild.groovy:330]: Yes, I will remove this command. Thx

File: ppg/pg_stat_monitor-autobuild.groovy (hunk 3 above)
Comment (adivinho) [reply ppg/pg_stat_monitor-autobuild.groovy:69]: Let's keep it in the current way. **_source_** is a Bash built-in function but we use **_sh_**.

**Review by EvgeniyPatlan — APPROVED** [review]

**Pull request merged**